sdk/src/test/java/io/customer/sdk/CustomerIOClientTest.kt

```diff
@@ -0,0 +1,198 @@
+import io.customer.base.utils.ActionUtils
+import io.customer.sdk.data.request.MetricEvent
+import io.customer.sdk.repository.IdentityRepository
+import io.customer.sdk.repository.PreferenceRepository
+import io.customer.sdk.repository.PushNotificationRepository
+import io.customer.sdk.repository.TrackingRepository
+import io.customer.sdk.utils.verifyError
+import io.customer.sdk.utils.verifySuccess
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mockito
+import org.mockito.Mockito.times
+import org.mockito.Mockito.verify
+import org.mockito.kotlin.any
+import org.mockito.kotlin.mock
+
+internal class CustomerIOClientTest {
+
+    private val preferenceRepository: PreferenceRepository = mock()
+    private val identityRepository: IdentityRepository = mock()
+    private val trackingRepository: TrackingRepository = mock()
+    private val pushNotificationRepository: PushNotificationRepository = mock()
+
+    private lateinit var customerIOClient: CustomerIOClient
+
+    @Before
+    fun setup() {
+        customerIOClient = CustomerIOClient(
+            identityRepository = identityRepository,
+            trackingRepository = trackingRepository,
+            pushNotificationRepository = pushNotificationRepository,
+            preferenceRepository = preferenceRepository
+        )
+    }
+
+    @Test
+    fun `verify client sends error when identify repo fails identifying`() {
+        Mockito.`when`(identityRepository.identify(any(), any()))
+            .thenReturn(
+                ActionUtils.getErrorAction(
+                    errorResult = ErrorResult(
+                        error = ErrorDetail(
+                            statusCode = StatusCode.InternalServerError
+                        )
+                    )
+                )
+            )
+
+        val result = customerIOClient.identify("identifier", mapOf()).execute()
+
+        verifyError(result, StatusCode.InternalServerError)
+    }
+
+    @Test
+    fun `verify client sends success when identify repo succeed in identifying`() {
+        Mockito.`when`(identityRepository.identify(any(), any()))
+            .thenReturn(ActionUtils.getEmptyAction())
+
+        val result = customerIOClient.identify("identifier", mapOf()).execute()
+
+        verifySuccess(result, Unit)
+    }
+
+    @Test
+    fun `verify when customer is identified then identifier is saved in prefs repo`() {
+        Mockito.`when`(identityRepository.identify(any(), any()))
+            .thenReturn(ActionUtils.getEmptyAction())
+
+        customerIOClient.identify("identifier", mapOf()).execute()
+
+        verify(preferenceRepository, times(1)).saveIdentifier(any())
+    }
+
+    @Test
+    fun `verify when customer identify is cleared its removed in prefs repo`() {
+        Mockito.`when`(preferenceRepository.getIdentifier()).thenReturn("identify")
+
+        customerIOClient.clearIdentify()
+
+        verify(preferenceRepository, times(1)).removeIdentifier(any())
```

Comment:
```suggestion
        verify(preferenceRepository, times(1)).removeIdentifier("identify")
```

To verify `clearIdentify` passes the expected value.